framework/include/actions/DetermineSystemType.h

```diff
@@ -28,6 +28,9 @@ class DetermineSystemType : public MooseObjectAction
   DetermineSystemType(InputParameters parameters);
 
   virtual void act() override;
+
+private:
+  InputParameters _action_params;
```

Comment:
OK, then I might suggest that you move _action_params to just a local variable. You only need it during act() so there's no need to store it as an instance variable.